packages/gatsby-plugin-sass/src/gatsby-node.js

```diff
@@ -1,64 +1,57 @@
-  const cssModulesConfDev = `${cssModulesConf}&sourceMap&localIdentName=[name]---[local]---[hash:base64:5]`
+exports.modifyWebpackConfig = ({ boundActionCreators, stage, rules, plugins, loaders }) => {
+  const { setWebpackConfig } = boundActionCreators
+  const PRODUCTION = stage !== `develop`
 
-  switch (stage) {
-    case `develop`: {
-      config.loader(`sass`, {
-        test: /\.s(a|c)ss$/,
-        exclude: /\.module\.s(a|c)ss$/,
-        loaders: [`style`, `css`, `sass`],
-      })
+  const sassLoader = {
+    loader: require.resolve(`sass-loader`),
+    options: { sourceMap: !PRODUCTION },
+  }
 
-      config.loader(`sassModules`, {
-        test: /\.module\.s(a|c)ss$/,
-        loaders: [`style`, cssModulesConfDev, `sass`],
-      })
-      return config
-    }
-    case `build-css`: {
-      config.loader(`sass`, {
-        test: /\.s(a|c)ss$/,
-        exclude: /\.module\.s(a|c)ss$/,
-        loader: ExtractTextPlugin.extract([`css?minimize`, `sass`]),
-      })
+  const sassRule = {
+    test: /\.s(a|c)ss$/,
+    exclude: /\.module\.s(a|c)ss$/,
+    use: plugins.extractText.extract({
+      fallback: loaders.style,
+      use: [loaders.css({ importLoaders: 1 }), loaders.postcss(), sassLoader],
+    }),
+  }
+  const sassRuleModules = {
+    test: /\.module\.s(a|c)ss$/,
+    use: plugins.extractText.extract({
+      fallback: loaders.style,
+      use: [
+        loaders.css({ modules: true, importLoaders: 1 }),
+        loaders.postcss(),
+        sassLoader,
+      ],
+    }),
+  }
 
-      config.loader(`sassModules`, {
-        test: /\.module\.s(a|c)ss$/,
-        loader: ExtractTextPlugin.extract(`style`, [cssModulesConf, `sass`]),
-      })
-      return config
-    }
-    case `build-html`: {
-      config.loader(`sass`, {
-        test: /\.s(a|c)ss$/,
-        exclude: /\.module\.s(a|c)ss$/,
-        loader: `null`,
-      })
+  let configRules = []
 
-      config.loader(`sassModules`, {
-        test: /\.module\.s(a|c)ss$/,
-        loader: ExtractTextPlugin.extract(`style`, [cssModulesConf, `sass`]),
-      })
-      return config
-    }
-    case `build-javascript`: {
-      config.loader(`sass`, {
-        test: /\.s(a|c)ss$/,
-        exclude: /\.module\.s(a|c)ss$/,
-        loader: `null`,
-      })
+  switch (stage) {
+    case `develop`:
+    case `build-css`:
+    case `build-javascript`:
+      configRules = configRules.concat([sassRule, sassRuleModules])
+      break
 
-      config.loader(`sassModules`, {
-        test: /\.module\.s(a|c)ss$/,
-        loader: ExtractTextPlugin.extract(`style`, [cssModulesConf, `sass`]),
-      })
-      return config
-    }
-    default: {
-      return config
-    }
+    case `build-html`:
+    case `develop-html`:
+      configRules = configRules.concat([
+        {
+          ...sassRule,
+          use: loaders.null,
+        },
+        sassRuleModules,
+      ])
+      break
   }
+
+  setWebpackConfig({
+    module: {
+      rules: configRules,
+    },
+  })
```

Comment:
yes, plugins execute and if they call the action you get an immediate update to the config in the store. I switched the `config` arg here to function so that you could get the most recent config from the store if needed.  Theoretically plugins can clobber each other's settings, but that's not any different really then before (or really avoidable for that matter)